rootston/virtual_keyboard.c

```diff
@@ -0,0 +1,29 @@
+#define _POSIX_C_SOURCE 199309L
+
+#include <wlr/util/log.h>
+#include <wlr/types/wlr_virtual_keyboard.h>
+#include "rootston/virtual_keyboard.h"
+#include "rootston/seat.h"
+
+static void handle_virtual_keyboard_destroy(struct wl_listener *listener,
+		void *data) {
+	// Not removing keyboard from seat, wlr_keyboard destroy handler will take
+	// care of it
+}
+
+void handle_virtual_keyboard(struct wl_listener *listener, void *data) {
+	struct roots_desktop *desktop =
+		wl_container_of(listener, desktop, virtual_keyboard_new);
+	struct wlr_virtual_keyboard *keyboard = data;
+
+	struct roots_seat *seat = input_seat_from_wlr_seat(desktop->server->input,
+		keyboard->seat);
+	if (!seat) {
+		wlr_log(L_ERROR, "could not find roots seat");
+		return;
+	}
+
+	roots_seat_add_device(seat, &keyboard->input_device);
+	wl_signal_add(&keyboard->destroy, &desktop->virtual_keyboard_destroy);
+	desktop->virtual_keyboard_destroy.notify = handle_virtual_keyboard_destroy;
+}
```

Comment:
Why register it when ```handle_virtual_keyboard_destroy``` is a noop?

Reply:
That's vestigial from an earlier version - removing.